docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,727 @@
+    minimumSize: 20Gi
+    providers:
+    - name: foo
+      purpose: etcd-main
+  networks: # Seed and Shoot networks must be disjunct
+    nodes: 10.240.0.0/16
+    pods: 10.241.128.0/17
+    services: 10.241.0.0/17
+  # Shoot default networks, see also https://github.com/gardener/gardener/issues/895.
+  # shootDefaults:
+  #   pods: 100.96.0.0/11
+  #   services: 100.64.0.0/13
+  taints:
+  - key: seed.gardener.cloud/protected
+  - key: seed.gardener.cloud/invisible
+  blockCIDRs:
+  - 169.254.169.254/32
+status:
+  conditions:
+  - lastTransitionTime: "2020-07-14T19:16:42Z"
+    lastUpdateTime: "2020-07-14T19:18:17Z"
+    message: all checks passed
+    reason: Passed
+    status: "True"
+    type: Available
+  gardener:
+    id: 4c9832b3823ee6784064877d3eb10c189fc26e98a1286c0d8a5bc82169ed702c
+    name: gardener-controller-manager-7fhn9ikan73n-7jhka
+    version: 1.0.0
+  observedGeneration: 1
+```
+
+### `SecretBinding` resource
+
+Special note: No modifications needed compared to the current `garden.sapcloud.io/v1beta1.SecretBinding` resource.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret1
+  namespace: garden-core
+type: Opaque
+data:
+  # <some-provider-specific data keys>
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-alicloud/example/30-infrastructure.yaml#L14-L15
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/example/30-infrastructure.yaml#L9-L10
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-azure/example/30-infrastructure.yaml#L14-L17
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-gcp/example/30-infrastructure.yaml#L14
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-openstack/example/30-infrastructure.yaml#L15-L18
+  # https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-packet/example/30-infrastructure.yaml#L14-L15
+  #
+  # If you use your own domain (not the default domain of your landscape) then you have to add additional keys to this secret.
+  # The reason is that the DNS management is not part of the Gardener core code base but externalized, hence, it might use other
+  # key names than Gardener itself.
+  # The actual values here depend on the DNS extension that is installed to your landscape.
+  # For example, check out https://github.com/gardener/external-dns-management and find a lot of example secret manifests here:
+  # https://github.com/gardener/external-dns-management/tree/master/examples
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: SecretBinding
+metadata:
+  name: secretbinding1
+  namespace: garden-core
+secretRef:
+  name: secret1
+# namespace: namespace-other-than-'garden-core' // optional
+quotas: []
```

Comment:
Why the `SecretBinding` should know about `Quotas`?
Shouldn't be the other way around? After all the Quotas are only needed for validation of request.

Reply:
Any redesign in that regard will cause a rewrite/refactoring of the current `quotavalidator` admission plugin. I don't want to do this as part of #308. Please file a separate PR with a proposal how a better API + admission implementation could look like.